### PR TITLE
Clean up Kinesis doc

### PR DESCRIPTION
**Pull request by ektravel**
### Description

This PR:
- Adds "Default" column to tables
- Removes most of future tense
- Collapses examples using `<details><summary></summary></details>`
- Changes heading from title case to sentence case
- Adds code font where applicable
- Removes the link to [PR 5102](https://github.com/apache/druid/pull/5102) which has already been merged
- Consolidates information
- Removed redundant information
- places intermediateHandoffPeriod after maxTotalRows
- documents skipBytesInMemoryOverheadCheck

This PR does not address the usage of passive voice. That can be done as a separate PR.

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 1]
```diff
@@ -408,84 +403,82 @@ If the `useListShards` flag is set to `false`, you need following permissions:
...
   last processed sequence number if continuing, or starting from the beginning or ending of the stream if this is a new stream).
-  2) Discover any running indexing tasks that are writing to the supervisor's datasource and adopt them if they match
+2. Discover any running indexing tasks that are writing to the supervisor's datasource and adopt them if they match
   the supervisor's configuration, else signal them to stop.
-  3) Send a status request to each supervised task to update our view of the state of the tasks under our supervision.
-  4) Handle tasks that have exceeded `taskDuration` and should transition from the reading to publishing state.
-  5) Handle tasks that have finished publishing and signal redundant replica tasks to stop.
-  6) Handle tasks that have failed and clean up the supervisor's internal state.
-  7) Compare the list of healthy tasks to the requested `taskCount` and `replicas` configurations and create additional tasks if required.
+3. Send a status request to each supervised task to update our view of the state of the tasks under our supervision.
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:440]:
```suggestion
3. Send a status request to each supervised task to update the view of the state of the tasks under supervision.
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 2]
```diff
@@ -408,84 +403,82 @@ If the `useListShards` flag is set to `false`, you need following permissions:
...
 
-### Suspending and Resuming Supervisors
+### Suspending and resuming supervisors
 
-You can suspend and resume a supervisor using `POST /druid/indexer/v1/supervisor/<supervisorId>/suspend` and `POST /druid/indexer/v1/supervisor/<supervisorId>/resume`, respectively.
+To suspend a supervisor, send a `POST` request to `/druid/indexer/v1/supervisor/<supervisorId>/suspend`.
+To resume a supervisor, send a `POST` request to `/druid/indexer/v1/supervisor/<supervisorId>/resume`.
 
-Note that the supervisor itself will still be operating and emitting logs and metrics,
+The supervisor itself will still be operating and emitting logs and metrics,
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:470]:
```suggestion
The supervisor still operates and emits logs and metrics,
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 3]
```diff
@@ -408,84 +403,82 @@ If the `useListShards` flag is set to `false`, you need following permissions:
...
 it will just ensure that no indexing tasks are running until the supervisor is resumed.
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:471]:
```suggestion
and it ensures that no indexing tasks are running until the supervisor resumes.
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 4]
```diff
@@ -501,34 +494,31 @@ enables you to recover from this condition.
...
 publishing their segments. This supervisor will still exist in the metadata store and its history may be retrieved
 with the supervisor history API, but will not be listed in the 'get supervisors' API response nor can its configuration
 or status report be retrieved. The only way this supervisor can start again is by submitting a functioning supervisor
 spec to the create API.
 
-### Capacity Planning
+### Capacity planning
 
-Kinesis indexing tasks run on MiddleManagers and are thus limited by the resources available in the MiddleManager
+Kinesis indexing tasks run on MiddleManagers and are limited by the resources available in the MiddleManager
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:508]:
```suggestion
Kinesis indexing tasks run on Middle Managers and are limited by the resources available in the Middle Manager
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 5]
```diff
@@ -501,34 +494,31 @@ enables you to recover from this condition.
...
-in data loss (assuming the tasks run before Kinesis purges those sequence numbers).
-
-A running task will normally be in one of two states: *reading* or *publishing*. A task will remain in reading state for
-`taskDuration`, at which point it will transition to publishing state. A task will remain in publishing state for as long
-as it takes to generate segments, push segments to deep storage, and have them be loaded and served by a Historical process
+shared across all types of indexing tasks, so you should plan your worker capacity to handle your total indexing load, such as batch processing, streaming tasks, and merging tasks. If your workers run out of capacity, Kinesis indexing tasks queue and wait for the next available worker. This may cause queries to return partial results but will not result in data loss (assuming the tasks run before Kinesis purges those sequence numbers).
+
+A running task can be in one of two states: reading or publishing. A task remains in reading state for
+`taskDuration`, at which point it transitions to publishing state. A task remains in publishing state for as long
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:514]:
```suggestion
the period defined in `taskDuration`, at which point it transitions to publishing state. A task remains in publishing state for as long
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 6]
```diff
@@ -501,34 +494,31 @@ enables you to recover from this condition.
...
+as it takes to generate segments, push segments to deep storage, and have them loaded and served by a Historical process
 (or until `completionTimeout` elapses).
 
 The number of reading tasks is controlled by `replicas` and `taskCount`. In general, there will be `replicas * taskCount`
-reading tasks, the exception being if taskCount > {numKinesisShards} in which case {numKinesisShards} tasks will
+reading tasks, the exception being if `taskCount > {numKinesisShards}` in which case `{numKinesisShards}` tasks will
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:519]:
```suggestion
reading tasks. An exception occurs if `taskCount > {numKinesisShards}` in which case `{numKinesisShards}` tasks are
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 7]
```diff
@@ -501,34 +494,31 @@ enables you to recover from this condition.
...
 be used instead. When `taskDuration` elapses, these tasks will transition to publishing state and `replicas * taskCount`
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:520]:
```suggestion
used instead. When `taskDuration` elapses, these tasks transition to publishing state and `replicas * taskCount`
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 8]
```diff
@@ -501,34 +494,31 @@ enables you to recover from this condition.
...
-new reading tasks will be created. Therefore, to allow for reading tasks and publishing tasks to run concurrently, there
+new reading tasks will be created. To allow for reading tasks and publishing tasks to run concurrently, there
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:521]:
```suggestion
new reading tasks are created. To allow for reading tasks and publishing tasks to run concurrently, there
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 9]
```diff
@@ -537,59 +527,54 @@ workerCapacity = 2 * replicas * taskCount
 
 This value is for the ideal situation in which there is at most one set of tasks publishing while another set is reading.
 In some circumstances, it is possible to have multiple sets of tasks publishing simultaneously. This would happen if the
-time-to-publish (generate segment, push to deep storage, loaded on Historical) > `taskDuration`. This is a valid
+time-to-publish (generate segment, push to deep storage, loaded on Historical) is greater than `taskDuration`. This is a valid
 scenario (correctness-wise) but requires additional worker capacity to support. In general, it is a good idea to have
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:531]:
```suggestion
and correct scenario but requires additional worker capacity to support. In general, it is a good idea to have
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 10]
```diff
@@ -537,59 +527,54 @@ workerCapacity = 2 * replicas * taskCount
...
-supervisor will create a new set of tasks. In this way, the supervisors are persistent across Overlord restarts and
+configured metadata database. There can only be a single supervisor per datasource, and submitting a second spec for
+the same datasource overwrites the previous one.
+
+When an Overlord gains leadership, either by being started or as a result of another Overlord failing, it spawns
+a supervisor for each supervisor spec in the metadata database. The supervisor then discovers running Kinesis indexing
+tasks and attempts to adopt them if they are compatible with the supervisor's configuration. If they are not
+compatible because they have a different ingestion spec or shard allocation, the tasks are killed and the
+supervisor creates a new set of tasks. In this way, the supervisors are persistent across Overlord restarts and
 fail-overs.
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:545]:
```suggestion
failovers.
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 11]
```diff
@@ -537,59 +527,54 @@ workerCapacity = 2 * replicas * taskCount
...
 In this way, configuration changes can be applied without requiring any pause in ingestion.
 
-### Deployment Notes
-
-#### On the Subject of Segments
+### Deployment notes
 
-Each Kinesis Indexing Task puts events consumed from Kinesis Shards assigned to it in a single segment for each segment
-granular interval until maxRowsPerSegment, maxTotalRows or intermediateHandoffPeriod limit is reached. At this point, a new shard
+Each Kinesis Indexing Task puts events consumed from Kinesis shards assigned to it in a single segment for each segment
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:563]:
```suggestion
Each Kinesis indexing task puts events consumed from Kinesis shards assigned to it in a single segment for each segment
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 12]
```diff
@@ -638,21 +623,21 @@ To enable this feature, set `deaggregate` to true in your `ioConfig` when submit
...
 The early shutdowns and task failures are expected. They occur because the supervisor updates the shard to task group mappings as shards are closed and fully read. This ensures that tasks are not running
 with an assignment of closed shards that have been fully read and balances distribution of active shards across tasks.
 
-This window with early task shutdowns and possible task failures will conclude when:
+This window with early task shutdowns and possible task failures concludes when:
 
-- All closed shards have been fully read and the Kinesis ingestion tasks have published the data from those shards, committing the "closed" state to metadata storage
-- Any remaining tasks that had inactive shards in the assignment have been shutdown (these tasks would have been created before the closed shards were completely drained)
+- All closed shards have been fully read and the Kinesis ingestion tasks have published the data from those shards, committing the "closed" state to metadata storage.
+- Any remaining tasks that had inactive shards in the assignment have been shutdown. These tasks would have been created before the closed shards were completely drained.
```
Comment (writer-jill) [inline docs/development/extensions-core/kinesis-ingestion.md:634]:
```suggestion
- Any remaining tasks that had inactive shards in the assignment have been shut down. These tasks would have been created before the closed shards were completely drained.
```

**Review by writer-jill — CHANGES_REQUESTED** [review]
Several small suggestions! Looks good

**Review by writer-jill — CHANGES_REQUESTED** [review]
Several small suggestions! Looks good

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 13]
```diff
@@ -23,20 +23,15 @@ sidebar_label: "Amazon Kinesis"
...
-- ensure that scalability and replication requirements are maintained.
-
-To use the Kinesis indexing service, load the `druid-kinesis-indexing-service` core Apache Druid extension (see
-[Including Extensions](../../configuration/extensions.md#loading-extensions)).
+To use the Kinesis indexing service, load the `druid-kinesis-indexing-service` core Apache Druid extension. See [Loading extensions](../../configuration/extensions.md#loading-extensions) for more information.
 
 > Before you deploy the Kinesis extension to production, read the [Kinesis known issues](#kinesis-known-issues).
 
-## Submitting a Supervisor Spec
+## Submitting a supervisor spec
```
[the remark this answers is not in the document]
Comment (ektravel) [reply docs/development/extensions-core/kinesis-ingestion.md:32]: 100% agree! I will save it for a different PR.

**Review by writer-jill — APPROVED** [review]
LGTM

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 14]
```diff
@@ -23,20 +23,15 @@ sidebar_label: "Amazon Kinesis"
...
 
-To use the Kinesis indexing service, load the `druid-kinesis-indexing-service` extension on both the Overlord and the MiddleManagers. Druid starts a supervisor for a dataSource when you submit a supervisor spec. Submit your supervisor spec to the following endpoint:
+To use the Kinesis indexing service, load the `druid-kinesis-indexing-service` extension on both the Overlord and the Middle Managers. Druid starts a supervisor for a datasource when you submit a supervisor spec. Submit your supervisor spec to the following endpoint:
```
Comment (techdocsmith) [inline docs/development/extensions-core/kinesis-ingestion.md:34]: nit: link to supervisors API doc?

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 15]
```diff
@@ -46,7 +41,10 @@ For example:
 curl -X POST -H 'Content-Type: application/json' -d @supervisor-spec.json http://localhost:8090/druid/indexer/v1/supervisor
```
Comment (techdocsmith) [inline docs/development/extensions-core/kinesis-ingestion.md:41]: imo it would be better to use a minimal inline supervisor spec in the payload rather than to link to a file. Using tabbed cURL & HTTP examples. See the sample request here: https://github.com/apache/druid/pull/14492/

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 16]
```diff
@@ -269,100 +273,91 @@ For more information, see [Data formats](../../ingestion/data-formats.md). You c
...
-|`offsetFetchPeriod`|ISO8601 period|How often the supervisor queries Kinesis and the indexing tasks to fetch current offsets and calculate lag. If the user-specified value is below the minimum value (`PT5S`), the supervisor ignores the value and uses the minimum value instead.|no (default == PT30S, min == PT5S)|
-|`useListShards`|Boolean|Indicates if `listShards` API of AWS Kinesis SDK can be used to prevent `LimitExceededException` during ingestion. Please note that the necessary `IAM` permissions must be set for this to work.|no (default == false)|
+The `tuningConfig` parameter is optional. If you don't specify `tuningConfig`, Druid uses default parameters.
```
Comment (techdocsmith) [inline docs/development/extensions-core/kinesis-ingestion.md:276]: `tuningConfig` is an object (JSON object), not a parameter. If you don't specify... Druid uses the defaults or the default configurations.

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 17]
```diff
@@ -269,100 +273,91 @@ For more information, see [Data formats](../../ingestion/data-formats.md). You c
...
-|`type`|String|Must be `roaring`.|yes|
-
-For Concise bitmaps:
-
-|Field|Type|Description|Required|
-|-----|----|-----------|--------|
-|`type`|String|Must be `concise`.|yes|
+|**Compression scheme**|**Field**|**Type**|**Description**|**Required**|
+|----------------------|---------|--------|---------------|------------|
+|Roaring|`type`|String|Must be `roaring`.|Yes|
```
Comment (techdocsmith) [inline docs/development/extensions-core/kinesis-ingestion.md:326]: The descriptions here are not doing a lot of work. We should either link out to help folks understand how to choose a compression scheme or add some info so people know what to choose. Also is one or the other of these the default. (This might be for another PR)

**Review by techdocsmith — COMMENTED** [review]
Some initial comments

File: docs/development/extensions-core/kinesis-ingestion.md (hunk 16 above)
Comment (ektravel) [reply docs/development/extensions-core/kinesis-ingestion.md:276]: Updated

**Review by writer-jill — APPROVED** [review]
Just a couple of suggestions - looks good!

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 18]
```diff
@@ -23,154 +23,259 @@ sidebar_label: "Amazon Kinesis"
...
-- ensure that scalability and replication requirements are maintained.
+This topic contains configuration reference information for the Kinesis indexing service supervisor for Apache Druid.
 
-To use the Kinesis indexing service, load the `druid-kinesis-indexing-service` core Apache Druid extension (see
-[Including Extensions](../../configuration/extensions.md#loading-extensions)).
+## Setup
 
-> Before you deploy the Kinesis extension to production, read the [Kinesis known issues](#kinesis-known-issues).
+To use the Kinesis indexing service, you must first load the `druid-kinesis-indexing-service` core extension on both the Overlord and the Middle Manager. See [Loading extensions](../../configuration/extensions.md#loading-extensions) for more information.
+We recommend that you review the [Kinesis known issues](#kinesis-known-issues) before deploying the `druid-kinesis-indexing-service` extension to production.
```
Comment (techdocsmith) [inline docs/development/extensions-core/kinesis-ingestion.md:33]:
```suggestion
Review the [Kinesis known issues](#kinesis-known-issues) before deploying the `druid-kinesis-indexing-service` extension to production.
```

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 19]
```diff
@@ -23,154 +23,259 @@ sidebar_label: "Amazon Kinesis"
...
+The following example shows how to submit a supervisor spec for a stream with the name `KinesisStream`.
+In this example, `http://SERVICE_IP:SERVICE_PORT` is a placeholder for the server address of deployment and the service port.
 
-```json
-{
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--cURL-->
+```shell
+curl -X POST "http://SERVICE_IP:SERVICE_PORT/druid/indexer/v1/supervisor" \
```
Comment (techdocsmith) [inline docs/development/extensions-core/kinesis-ingestion.md:65]: normally we're running all HTTP API calls through the router. Retest it with this :
`curl -X POST "http://SERVICE_IP:SERVICE_PORT/druid/router/v1/supervisor"`

`

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 20]
```diff
@@ -23,154 +23,259 @@ sidebar_label: "Amazon Kinesis"
...
+        "rollup": false,
+        "segmentGranularity": "hour"
+      }
+    }
+  }
+}'
+```
+<!--HTTP-->
+```HTTP
+POST /druid/indexer/v1/supervisor
```
Comment (techdocsmith) [inline docs/development/extensions-core/kinesis-ingestion.md:140]: same comment as line 65 wrt/ router vs indexer

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 21]
```diff
@@ -23,154 +23,259 @@ sidebar_label: "Amazon Kinesis"
...
+|`taskCountMax`|Maximum number of Kinesis ingestion tasks. Must be greater than or equal to `taskCountMin`. If greater than `{numKinesisShards}`, Druid sets the maximum number of reading tasks to `{numKinesisShards}` and ignores `taskCountMax`.|Yes||
+|`taskCountMin`|Minimum number of Kinesis ingestion tasks. When you enable the auto scaler, Druid ignores the value of `taskCount` in `IOConfig` and uses `taskCountMin` for the initial number of tasks to launch.|Yes||
+|`minTriggerScaleActionFrequencyMillis`|Minimum time interval between two scale actions.| No|600000|
+|`autoScalerStrategy`|The algorithm of `autoScaler`. Druid only supports the `lagBased` strategy. See [Lag based autoscaler strategy related properties](#lag-based-autoscaler-strategy-related-properties) for more information.|No|Defaults to `lagBased`.|
+
+### Lag based autoscaler strategy related properties
+
+Unlike the Kafka
+indexing service, Kinesis reports lag metrics measured in time difference in milliseconds between the current sequence number and latest sequence number, rather than message count.
```
Comment (techdocsmith) [inline docs/development/extensions-core/kinesis-ingestion.md:258]:
```suggestion
Unlike the Kafka indexing service, Kinesis reports lag metrics measured in time difference in milliseconds between the current sequence number and latest sequence number, rather than message count.
```

**Review by techdocsmith — CHANGES_REQUESTED** [review]

**Review by techdocsmith — CHANGES_REQUESTED** [review]
See comments for requested changes.

**Review by techdocsmith — APPROVED** [review]
LGTM
